modules/service/challenge-service.js

```diff
@@ -80,7 +75,8 @@ class ChallengeService {
         blockIndex, encryptedObject,
         blockSizeInBytes = constants.DEFAULT_CHALLENGE_BLOCK_SIZE_BYTES,
     ) {
-        const answer = this.getBlockFromObject(encryptedObject, blockIndex, blockSizeInBytes);
+        const sortedObject = utilities.sortedStringify(encryptedObject, true);
```

Comment:
Are we sure that sortedStringify is valid function for sorting blocks? There is also sortObjectRecursively function.

Reply:
This is the sort method used inside getBlockFromObject, i just moved it outside so that we can avoid unnecessary sorting.